src/raft.rs

```diff
@@ -381,6 +387,14 @@ impl<T: Storage> Raft<T> {
         self.skip_bcast_commit = skip;
     }
 
+    #[inline]
+    fn max_uncommitted_size(&self) -> usize {
+        if self.state == StateRole::Leader {
```

Comment:
Can you please document why this conditional is needed?

Reply:
Main reason was that the Go change this is all based on decided they only need to apply this limit at the current leader.

I had thought this particular accessor would wind up getting used in more than one place. That turns out to not be the case, so it could be just hoisted into that function.